### DISCUSSION TITLE
Changing Color Theme for IDE

### DISCUSSION THREAD

**Discussion by ruffsl** [community/community#27214] [Codespaces]
<p>I used the search menu in the settings window, and didn’t see an option to change the default editor theme from light to dark. I prefer the dark theme for accessibility and high contrast visibility.</p>

**Comment by ruffsl** [comment, marked as answer]
<p>It looks like the option to select alternate themes can only be found via <code>File&gt;Preferences&gt;Color Theme</code> or <code>Ctrl+K Ctrl+T</code> and not under the <code>Setting</code> window <code>File&gt;Preferences&gt;Settings</code>. Odd.</p>

**Comment by sanketplus** [comment]
<p>+1, theme is not changing. And stays the same white theme</p>

**Comment by lostintangent** [comment]
<p><a class="mention" href="/ruffsl">@ruffsl</a> So it sounds like you were able to figure out how to change the theme? In addition to the options you mentioned, you can also click the gear icon in the bottom left of the editor, then select <code>Color Theme</code>.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5400f6084769da67f96f584ff73a5cf26806e6e1.png" data-download-href="/uploads/short-url/bZ8akmstCfR5Bg2sCs5BESiCyzL.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/116461/181102258-06281de1-5c90-4563-a9b1-05aa5ef2f7e6.png" alt="image" data-base62-sha1="bZ8akmstCfR5Bg2sCs5BESiCyzL" width="442" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5400f6084769da67f96f584ff73a5cf26806e6e1_2_442x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5400f6084769da67f96f584ff73a5cf26806e6e1_2_663x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/5400f6084769da67f96f584ff73a5cf26806e6e1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5400f6084769da67f96f584ff73a5cf26806e6e1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">710×802 51.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><a class="mention" href="/sanketplus">@sanketplus</a> If you follow the above instructions, and select the <code>GitHub Dark</code> theme, it should persist that selection across your Codespace sessions. Are you not seeing that behavior?</p>

**Comment by sanketplus** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/e/ee9bde35ee68724c3ea49f2e6dac6e9b9fa69289.png" data-download-href="/uploads/short-url/y2PAbvx1pr3mep74jVljw5JAOSt.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6016802/181102260-d8f04923-0748-48f0-b1ee-0831f0a587be.png" alt="image" data-base62-sha1="y2PAbvx1pr3mep74jVljw5JAOSt" width="446" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/e/ee9bde35ee68724c3ea49f2e6dac6e9b9fa69289_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">488×547 20.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>  don’t see that option

**Comment by lostintangent** [comment]
<p>If you click “Color Theme”, that should display a drop down list of the installed themes. From there, you should see “GitHub Dark” as an option.</p>

**Comment by MOlausson** [comment]
<p>It there a way to save the preferences? I tried to sync settings with my local VS Code but so far with no luck. Perhaps there’s a config file setting for it?</p>

**Comment by matthewisabel** [comment]
<p>You can turn on settings sync by going to your persona coin in the lower left. When you enable settings sync, this will turn on settings sync across VS Code Insiders (so all codespaces and VS Code Insiders client). In the near-term we’ll move Codespaces off of specific Insiders builds and on to VS Code. At the moment, settings sync is still in preview and by channel, so it sync across Insiders and VS Code stable separately.</p>
<p>tl;dr If you enable settings sync it will sync across codespaces/insiders. In the future is will sync across VS Code stable once Codespaces is also running that by default.</p>

**Comment by MOlausson** [comment]
<p>Thanks for pointing out I need to use the Insiders build. Now I enabled sync from both the web (codespaces) and desktop VS Code Insider but it’s flaky. If I change the theme in the Codespaces session it syncs with the desktop. If I change in the desktop it syncs in the Codespace session but then jumps over to the default Light+ theme.</p>
<p>Also when I start a new Codespace I have to enable sync, it would be nice if it could keep my last state for that.</p>
<p>Here’s the error I see in the synclog in my Codespace session:</p>
<pre><code class="lang-auto">[2020-09-07 22:09:50.601] [userDataSync.log] [error] Error: Unable to resolve non-existing file 'vscode-userdata:/User/sync/60ae0a5c/globalState/preview/globalState.json'
    at S.resolve (https://solidifymo-dotnetweb-xrhj.github.dev/workbench-page/web-standalone/insider-b6d4fec/out/vs/workbench/workbench.web.api.js:1672:765)
    at async e.doGenerateSyncResourcePreview (https://solidifymo-dotnetweb-xrhj.github.dev/workbench-page/web-standalone/insider-b6d4fec/out/vs/workbench/workbench.web.api.js:2817:2)
[2020-09-07 22:09:50.617] [userDataSync.log] [error] globalState: Unable to resolve non-existing file 'vscode-userdata:/User/sync/60ae0a5c/globalState/preview/globalState.json'
[2020-09-07 22:09:50.820] [userDataSync.log] [info] Extensions: No changes found during synchronizing extensions.
</code></pre>

**Comment by trevorhreed** [comment]
<p>I just got access to the Codespaces beta. I’m trying to change the theme. I can access the option using the File menu, the gear icon, or in Settings, and I can select the dark theme with no problem and no errors, but the theme never actually changes.</p>

**Comment by sr229** [comment]
<p>The reason for this is because the resources never properly loaded. You may have to refresh or do a hard refresh while emptying the cache.</p>

**Comment by gabeio** [comment]
<p>So I was having this problem as well within the latest version of Firefox (I have yet to get it to work in FF at all just mentioning that I’m up to date on FF), when I use Chrome it works correctly.</p>
<p>Not only does the theme not load/work, code highlighting and file icons also do not work for FF as well.</p>
